components/brave_rewards/browser/database/publisher_info_database.cc

```diff
@@ -1132,6 +1129,57 @@ ledger::ServerPublisherInfoPtr PublisherInfoDatabase::GetServerPublisherInfo(
 
 // Other -------------------------------------------------------------------
 
+void PublisherInfoDatabase::RecordP3AStats(bool auto_contributions_on) {
+  if (!initialized_) {
+    return;
+  }
+
+  sql::Statement sql(
+      db_.GetCachedStatement(SQL_FROM_HERE,
+                             "SELECT category, COUNT(*) FROM contribution_info "
+                             "WHERE category in (?, ?, ?) GROUP BY 1"));
+
+  sql.BindInt(0, ledger::AUTO_CONTRIBUTE);
+  sql.BindInt(1, ledger::ONE_TIME_TIP);
+  sql.BindInt(2, ledger::RECURRING_TIP);
+
+  int auto_contributions = 0;
+  int tips = 0;
+  int queued_recurring = 0;
+
+  while (sql.Step()) {
+    const int count = sql.ColumnInt(1);
+    switch (sql.ColumnInt(0)) {
+    case ledger::AUTO_CONTRIBUTE:
+      auto_contributions = count;
+      break;
+    case ledger::ONE_TIME_TIP:
+      tips += count;
+      break;
+    case ledger::RECURRING_TIP:
+      queued_recurring = count;
+      break;
+    default:
+      NOTREACHED();
+    }
+  }
+
+  if (queued_recurring == 0) {
```

Comment:
not sure if this logic cover everything. So what will not cover is all new recurring donations that you added if you had recurring donation last month. For example: In Aug you had one recurring donation that went through. In Sep you added 10 of them. This new ones will not be recorded until Oct

Reply:
Can you please help us to fix it? Not sure if I can understand all the business logic without some advice :)

Reply:
sorry missed this one, let's take of offline